### ISSUE TITLE
Performance Optimization Suggestion

### ISSUE THREAD

**Issue by lossycrypt** [amagid/Factorio-Shield-Walls-Redux#1]
I haven't tested this mod in any larger setup, but your mod description mentions performance problems. So here's an idea: In some subversion of 0.16 a new event on_entity_damaged was introduced. Using this it should be possible to keep a list of all _damaged_ walls, instead of a list of _all_ walls. Which should reduce the interation requirement and would mean 0 performance impact in a base with no damaged walls.

**Comment by amagid** [comment]
Interesting point - I hadn't heard about that event yet, thanks! My one concern is that the API notes that the on_entity_damaged event is not called if the entity is damaged directly by a script, which may impact the mod's interaction with other mods. I'll make a branch for this and look into it. Thanks for the suggestion!

**Comment by lossycrypt** [comment]
True. If there's mods that damage walls via scripts at all i haven't heared of them. Endgame combat has a turrent (shockwave-somthing-turrent? forgot name) that if used in MP against an opposing forces walls might do this, but i'd call that a very edgy edge case. If you're concerned you could keep the list of all walls and iterate over it at a much slower pace than now just to be sure you don't miss anything. But that would ofc negate the "no out-of-combat performance impact" aspect.
